pandas/tests/reshape/merge/test_merge.py

```diff
@@ -428,6 +429,64 @@ def check2(exp, kwarg):
                 check1(exp_in, kwarg)
                 check2(exp_out, kwarg)
 
+    @pytest.mark.parametrize(
+        'join_col, val_col', list(itertools.product([
+            pd.Series([1], dtype='int64'),
+            pd.Series([1], dtype='Int64'),
+            pd.Series([1.23]),
+            pd.Series(['foo']),
+            pd.Series([True]),
+            pd.Series([pd.Timestamp('2018-01-01')]),
+            pd.Series([pd.Timestamp('2018-01-01', tz='US/Eastern')]),
+        ], repeat=2)),
+        ids=lambda x: x.dtype.name
+    )
+    def test_merge_empty_frame(self, join_col, val_col):
+        # GH 25183
+        df = pd.DataFrame({'a': join_col, 'b': val_col}, columns=['a', 'b'])
+        df_empty = df[:0]
+        exp = pd.DataFrame({
+            'b_x': pd.Series(dtype=df.dtypes['b']),
+            'a': pd.Series(dtype=df.dtypes['a']),
+            'b_y': pd.Series(dtype=df.dtypes['b']),
+        }, columns=['b_x', 'a', 'b_y'])
+        act = df_empty.merge(df, on='a')
+        assert_frame_equal(act, exp)
+
+    @pytest.mark.parametrize(
+        'join_col, val_col', list(itertools.product(
```

Comment:
you can make these both fixures, they will automatically be the product if you pass both

Reply:
done